go/ql/test/library-tests/semmle/go/frameworks/GoMicro/gomicro.ql

```diff
@@ -0,0 +1,7 @@
+import go
+
+query predicate serverRequests(DataFlow::ParameterNode node) { node instanceof GoMicro::Request }
+
+query predicate clientRequests(DataFlow::Node node) {
+  node instanceof GoMicro::ClientRequestUrlAsSink
+}
```

Comment:
Switch to inline expectations test
```suggestion
import TestUtilities.InlineExpectationsTest

module GoMicroTest implements TestSig {
  string getARelevantTag() { result = ["serverRequest", "clientRequest"] }

  predicate hasActualResult(Location location, string element, string tag, string value) {
    exists(DataFlow::Node node |
      node.hasLocationInfo(location.getFile().getAbsolutePath(), location.getStartLine(),
        location.getStartColumn(), location.getEndLine(), location.getEndColumn()) and
      (
        node instanceof GoMicro::Request and
        element = node.toString() and
        value = "\"" + node.toString() + "\"" and
        tag = "serverRequest"
        or
        node instanceof GoMicro::ClientRequestUrlAsSink and
        element = node.toString() and
        value = node.toString().replaceAll("/", "\\/") and
        tag = "clientRequest"
      )
    )
  }
}

import MakeTest<GoMicroTest>
```

Reply:
I didn't know about this pattern, I will use it from now on